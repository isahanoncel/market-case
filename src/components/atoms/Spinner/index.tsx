import React, { FC } from 'react';
import * as S from './Spinner.styled';

interface ISpinner {
  color?: string;
  size?: number;
  strokeWidth?: number;
  margin?: string;
}

const Spinner: FC<ISpinner> = ({
  color = 'Gray',
  size = 40,
  strokeWidth = 5,
  margin,
}) => (
  <S.Svg margin={margin} viewBox="0 0 50 50" width={size}>
    <S.Circle
      cx="25"
      cy={25}
      r={20}
      fill="none"
      strokeWidth={strokeWidth}
      stroke={color}
    />
  </S.Svg>
);

export default Spinner;
