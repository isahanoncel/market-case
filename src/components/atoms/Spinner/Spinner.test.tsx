import React from 'react';
import renderer from 'react-test-renderer';
import Spinner from '.';

test('Spinner snapshot testing', () => {
  const tree = renderer.create(<Spinner />).toJSON();
  expect(tree).toMatchSnapshot();
});
