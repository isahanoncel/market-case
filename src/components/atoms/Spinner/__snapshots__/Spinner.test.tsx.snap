// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner snapshot testing 1`] = `
.c0 {
  -webkit-animation: rotate 2s linear infinite;
  animation: rotate 2s linear infinite;
  z-index: 2;
  top: 50%;
  left: 50%;
}

.c1 {
  stroke-linecap: round;
  -webkit-animation: dash 1.5s ease-in-out infinite;
  animation: dash 1.5s ease-in-out infinite;
}

<svg
  className="c0"
  viewBox="0 0 50 50"
  width={40}
>
  <circle
    className="c1"
    cx="25"
    cy={25}
    fill="none"
    r={20}
    stroke="Gray"
    strokeWidth={5}
  />
</svg>
`;
